src/drawerbase.js

```diff
@@ -0,0 +1,291 @@
+        //idiomatic javascript options object as the only argument
+        var args  = arguments;
+
+        if( !$.isPlainObject( options ) ){
+            options = {
+                source:     args[ 0 ], // Reference to Viewer tile source.
+                viewport:   args[ 1 ], // Reference to Viewer viewport.
+                element:    args[ 2 ]  // Parent element.
+            };
+        }
+
+        $.console.assert( options.viewport, "[Drawer] options.viewport is required" );
+        $.console.assert( options.element, "[Drawer] options.element is required" );
+
+        if ( options.source ) {
+            $.console.error( "[Drawer] options.source is no longer accepted; use TiledImage instead" );
+        }
+
+        this.viewer = options.viewer;
+        this.viewport = options.viewport;
+        this.debugGridColor = typeof options.debugGridColor === 'string' ? [options.debugGridColor] : options.debugGridColor || $.DEFAULT_SETTINGS.debugGridColor;
+        this.options = options.options || {};
+
+        /**
+         * The parent element of this Drawer instance, passed in when the Drawer was created.
+         * The parent of {@link OpenSeadragon.DrawerBase#canvas}.
+         * @member {Element} container
+         * @memberof OpenSeadragon.DrawerBase#
+         */
+        this.container  = $.getElement( options.element );
+
+        // We force our container to ltr because our drawing math doesn't work in rtl.
+        // This issue only affects our canvas renderer, but we do it always for consistency.
+        // Note that this means overlays you want to be rtl need to be explicitly set to rtl.
+        this.container.dir = 'ltr';
+
+        // the first time this.canvas is accessed, implementations will create it
+        this.canvas.style.width     = "100%";
+        this.canvas.style.height    = "100%";
+        this.canvas.style.position  = "absolute";
+        $.setElementOpacity( this.canvas, this.viewer.opacity, true ); // set
+
+        // Allow pointer events to pass through the canvas element so implicit
+        //   pointer capture works on touch devices
+        $.setElementPointerEventsNone( this.canvas );
+        $.setElementTouchActionNone( this.canvas );
+
+        // explicit left-align
+        this.container.style.textAlign = "left";
+        this.container.appendChild( this.canvas );
+
+        this._checkForAPIOverrides();
+    }
+
+    get canvas(){
+        if(!this._renderingTarget){
+            this._renderingTarget = this._createDrawingElement();
+        }
+        return this._renderingTarget;
+    }
+    get element(){
+        $.console.error('Drawer.element is deprecated. Use Drawer.container instead.');
+        return this.container;
+    }
+
+    /**
+     * @property {String|undefined} type What type of drawer this is. Implementations should override this property.
+     */
+    getType(){
+        return undefined;
```

Comment:
I guess this could produce an error like the other functions that need overriding?

Reply:
Yeah I think that makes sense. I also realized the doc comment was not updated for it being a method instead of a getter.